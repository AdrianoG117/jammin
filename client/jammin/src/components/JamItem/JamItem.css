.single-jam {
  display: flex;
  width: 100%;
  background-color: rgba(0, 0, 0, 0.3);
  border-radius: 10px;
  margin: 10px;
  border: 1px solid whitesmoke;
  padding: 1rem;
}

.jam-info{
  display: flex;
  flex-direction: column;
  color: whitesmoke;
  width: 80%
}

.see-event {
display: flex;
justify-content: center;
align-items: center;
margin: 1rem;
height: 100%;
border-radius: 10px;
width: 20%
}

.see-event button {
  background-color: rgba(245,245,245,0.7);
  border-radius: 10px;
  height: 4rem;
  width: 8rem;
  font-size: 1rem;
  font-weight: bold
}

.see-event button:hover {
  color: whitesmoke;
  background-color: rgba(0, 0, 0, 0.4)
}