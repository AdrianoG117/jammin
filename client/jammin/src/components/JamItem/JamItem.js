import React from 'react';
import './JamItem.css';
import { Link } from 'react-router-dom';
import moment from 'moment';

function JamItem({ jam, highEvent, setHighEvent }) {
  return (
    <div
      onMouseEnter={() => {
        console.log('jamCoords : ', jam.locCords);
        setHighEvent(jam.locCords);
      }}
      onMouseLeave={() => {
        setHighEvent(null);
      }}
    >
      <div className="single-jam">
        <div className="jam-info">
          <h1>{moment(jam.date).format('MMM Do, h:mm a')}</h1>
          <h1>{jam.title}</h1>
          <div className="jam-loc">
            <img src="" alt="" />
            <p>{jam.location}</p>
          </div>
        </div>
        <div className="see-event">
          <Link
            to={{
              pathname: `/jams/${jam._id}`,
              state: {
                jam: jam,
              },
            }}
          >
            <button>SEE EVENT</button>
          </Link>
        </div>
      </div>
    </div>
  );
}

export default JamItem;
