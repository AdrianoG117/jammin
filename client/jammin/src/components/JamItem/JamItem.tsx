import {FunctionComponent} from "react";
import "./JamItem.css";
import { useHistory } from "react-router-dom";
import moment from "moment";
import {Jam} from "../../apiService/APIResponseTypes";

type cords = { lat: number
    lng: number} | null

interface IProps{
  jam: Jam
 setHighEvent: React.Dispatch<React.SetStateAction< cords>>
}


const JamItem:FunctionComponent<IProps> = ({ jam,setHighEvent }) => {
  const history = useHistory();
  return (
    <div
      onMouseEnter={() => {
        setHighEvent(jam.locCords);
      }}
      onMouseLeave={() => {
        setHighEvent(null);
      }}
    >
      <div className="single-jam">
        <div className="jam-info">
          <h1>{moment(jam.date).format("MMM Do, h:mm a")}</h1>
          <h1>{jam.title}</h1>
          <div className="jam-loc">
            <img src="" alt="" />
            <p>{jam.location}</p>
          </div>
        </div>
        <div className="see-event" >
            <button onClick={()=> history.push({
              pathname: `/jams/${jam._id}`,
              state: {
                jam:jam
              }
            })}>SEE EVENT</button>
          
        </div>
      </div>
    </div>
  );
}

export default JamItem;
