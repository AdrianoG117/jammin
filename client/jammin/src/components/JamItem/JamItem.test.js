import {render, screen} from "@testing-library/react";
import JamItem from "./JamItem";
import moment from "moment";

const mockProps = {
  jam: {
    city: "Miami, FL, USA",
    cityCords: {lat: 25.7616798, lng: -80.1917902},
    comingEvent: true,
    date: "2021-10-02T17:12",
    description: "Poop",
    host: "",
    languages: "English",
    locCords: {lat: 25.7616798, lng: -80.1917902},
    location: "Miami, FL, USA",
    messages: [],
    numOfParticipants: 1,
    pastEvent: false,
    title: "Test1",
  },
  setHighEvent: ()=>{},
  
};

describe("JamItem component", () => {
  test("should match the snapshot", () => {
    const {container} = render(<JamItem
      jam={mockProps.jam}
      setHighEvent={mockProps.setHighEvent}
      ></JamItem>);
      expect(container.firstChild).toMatchSnapshot();
  });
  });
  
  describe("prop value rendering", () => {
    beforeEach(() => {
      render(<JamItem
            jam={mockProps.jam}
            setHighEvent={mockProps.setHighEvent}
            ></JamItem>);
});
     test("rendered title should match jam prop", () => {
      screen.getByText(`${mockProps.jam.title}`);
     });
     test("rendered location should match jam prop", () => {
      screen.getByText(`${mockProps.jam.location}`);
     });
     test("rendered date should match jam prop", () => {
      screen.getByText(`${moment(mockProps.jam.date).format("MMM Do, h:mm a")}`);
     });    
  });
