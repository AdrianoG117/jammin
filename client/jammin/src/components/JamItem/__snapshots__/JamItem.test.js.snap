// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JamItem component should match the snapshot 1`] = `
<div>
  <div
    class="single-jam"
  >
    <div
      class="jam-info"
    >
      <h1>
        Oct 2nd, 5:12 pm
      </h1>
      <h1>
        Test1
      </h1>
      <div
        class="jam-loc"
      >
        <img
          alt=""
          src=""
        />
        <p>
          Miami, FL, USA
        </p>
      </div>
    </div>
    <div
      class="see-event"
    >
      <button>
        SEE EVENT
      </button>
    </div>
  </div>
</div>
`;
